### PR TITLE
Implemented Analyzer class

### PR DESCRIPTION
**Pull request by IreneYangClari**
Run the `python Analyzer.py <model_type> <data_didr>`. It will generate two reports under the report folder:
1. hist_scores_<model_type>.png
2. perf_report_<model_type>.png

### CODE REVIEW

File: analyzer/Analyzer.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        labels = [l[0] for l in label_text_tuple]
+        predicted_labels = [s[0] for s in predicted_label_score]
+        predicted_probs = [s[1] for s in predicted_label_score]
+        predicted_scores = [int(s[1]*100) for s in predicted_label_score] # scale to 0 to 100
+
+        ## plot the histogram of predicted scores
+        self._plot_hist(report_dir, predicted_scores)
+
+        ## calculate performance matrix and write to report file
+        self._report_performance(labels, predicted_labels, predicted_probs)
```
Comment (dhruva-clari) [inline analyzer/Analyzer.py:39]: report_dir, model_type are missing

Comment (IreneYangClari) [reply analyzer/Analyzer.py:39]: good catch. We will take change once Ankita's change is in. The prediction part will need to be changed.

File: analyzer/Analyzer.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            use_trained_model = True
+
+        if use_trained_model:
+            predictor = CustomModelPredictor(model_dir)
+        else:
+            predictor = BaselineModelPredictor(model_dir)
+
+        ## load up the test set and scoring the data set
+        with open(str(path_to_file)) as f:
+            lines = f.readline()
```
Comment (goswaank) [inline analyzer/Analyzer.py]: I think you meant readlines()

File: analyzer/Analyzer.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            ## data is in a format [[label, text]]
+            label_text_tuple = [self.__parse_line(line) for line in lines]
+
+        print("Evaluating model {} on {} size of data".format(model_type, len(label_text_tuple)))
+        predicted_label_score = [predictor.predict(t[1]) for t in label_text_tuple]
```
Comment (goswaank) [inline analyzer/Analyzer.py]: Add the preprocessing step here.

File: analyzer/Analyzer.py (hunk 2 above)
Comment (goswaank) [reply analyzer/Analyzer.py]: resolved

File: analyzer/Analyzer.py (hunk 3 above)
Comment (goswaank) [reply analyzer/Analyzer.py]: resolved

**Review by goswaank — APPROVED** [review]

**Pull request merged**